MapleServer2/Types/ItemStats.cs

```diff
@@ -109,28 +109,28 @@ public ItemStats(int itemId, int grade)
                 }
             }
 
-            if (ItemStatsMetadataStorage.GetRandomStat(itemId, out List<ItemOptions> randomList))
+            if (ItemStatsMetadataStorage.GetRandomBonus(itemId, out List<ItemOptions> randomBonusList))
             {
-                foreach (ItemOptions item in randomList)
+                foreach (ItemOptions options in randomBonusList)
                 {
-                    if (item.Grade != grade)
+                    if (options.Rarity != rarity)
                     {
                         continue;
                     }
-                    if (item.OptionNumPick == 0)
+                    if (options.Slots == 0)
                     {
                         continue;
                     }
-                    List<int> indexes = GetRandomOptions(item.OptionNumPick, item.Stats.Count);
+                    List<int> indexes = GetRandomOptions(options.Slots, options.Stats.Count);
                     foreach (int i in indexes)
                     {
-                        if (item.Stats[i].Value != 0)
+                        if (options.Stats[i].Value != 0)
```

Comment:
You could probably write a method to reuse this code, you're doing the same thing for basic attributes.